lib/remote/apilistener.cpp

```diff
@@ -1303,12 +1304,26 @@ bool ApiListener::RelayMessageOne(const Zone::Ptr& targetZone, const MessageOrig
 			bool isMaster = (currentZoneMaster == localEndpoint);
 
 			if (!isMaster && targetEndpoint != currentZoneMaster) {
+				if (currentTargetZone == parentZone) {
+					if (m_CurrentParentEndpoint.exchange(currentZoneMaster.get()) != currentZoneMaster.get()) {
+						Log(LogInformation, "ApiListener") << "Relaying messages for parent Zone '"
+							<< parentZone->GetName() << "' to Endpoint '" << currentZoneMaster->GetName() << "' now";
+					}
+				}
+
 				skippedEndpoints.push_back(targetEndpoint);
 				continue;
 			}
 
 			relayed = true;
 
+			if (currentTargetZone == parentZone) {
+				if (m_CurrentParentEndpoint.exchange(targetEndpoint.get()) != targetEndpoint.get()) {
+					Log(LogInformation, "ApiListener") << "Relaying messages for parent Zone '"
+						<< parentZone->GetName() << "' to Endpoint '" << targetEndpoint->GetName() << "' now";
+				}
```

Comment:
I don't think that endpoint caching is required or is there something that I'm missing? Apart from that, I would remove the ` now` suffix from the log message, it's kinda obvious that it's now, when you see that log entry.

Reply:
> or is there something that I'm missing?

Indeed, the thing I missed is that this log entry should only be logged initially and when the connected endpoint changes at some point. So, that should be fine!